torch/_refs/__init__.py

```diff
@@ -1298,14 +1298,14 @@ def _lcm(a: TensorLikeType, b: TensorLikeType):
     dtype = a.dtype
     promote_to_int = dtype in (torch.int8, torch.int16)
     if promote_to_int:
-        a = prims.convert_element_type(a, torch.int32)
-        b = prims.convert_element_type(b, torch.int32)
+        a = _maybe_convert_to_dtype(a, torch.int32)
```

Comment:
In this case the promotion will definitely occur because we know `a` has a different dtype than `torch.int32`

Reply:
Is this implementation correct thought? It seems odd that it's only checking the dtype of `a` -- what about the dtype of `b`? Also it could use more comments.

Reply:
Yes, I think so after reading this (originally investigated in https://github.com/pytorch/pytorch/pull/82950#issue-1331192753).

> lcm is meant to use integer maths, but the use of true_divide
introduces a promotion to float and thus a loss of precision.
This also introduces promoting low precision integers to int32 which
is required for 100% consistency with the C++ implementation.

Thanks!